main.py

```diff
@@ -1,228 +1,172 @@
-async def _eval(ctx, *, code):
-    """
-    Evaluates given code.
-    """
-    code = clean_code(code)
-
-    local_variables = {
-        "discord": discord,
-        "commands": commands,
-        "bot": bot,
-        "ctx": ctx,
-        "channel": ctx.channel,
-        "author": ctx.author,
-        "guild": ctx.guild,
-        "message": ctx.message,
-    }
-
-    stdout = io.StringIO()
-
-    try:
-        with contextlib.redirect_stdout(stdout):
-            exec(
-                f"async def func():\n{textwrap.indent(code, '    ')}", local_variables,
-            )
-
-            obj = await local_variables["func"]()
-            result = f"{stdout.getvalue()}\n-- {obj}\n"
-
-    except Exception as e:
-        result = "".join(format_exception(e, e, e.__traceback__))
-
-    pager = Pag(
-        timeout=180,
-        use_defaults=True,
-        entries=[result[i : i + 2000] for i in range(0, len(result), 2000)],
-        length=1,
-        prefix="```py\n",
-        suffix="```",
-    )
-
-    await pager.start(ctx)
+            await self.config.get_all()
+        except exceptions.PyMongoError as error:
+            log = 'An error occured while fetching the config: %s' % error
+            self.logger.error(log)
+        else:
+            self.logger.info('Database connection established.')
 
+    async def on_message(self, message):
+        # Ignore messages sent by bots.
+        if message.author.bot:
+            return None
 
-@bot.command()
-@commands.is_owner()
-async def dbbackup(ctx):
-    """Back up the database"""
-    await ctx.send("https://giphy.com/gifs/christmas-3P0oEX5oTmrkY")
+        guild_id = message.guild.id
 
-    backupDB = motor.motor_asyncio.AsyncIOMotorClient(config["mongo_url"]).backup
-    backupConfig = Document(backupDB, "config")
-    backupKeywords = Document(backupDB, "keywords")
-    backupQuiz = Document(backupDB, "quiz")
-    backupCode = Document(backupDB, "code")
-    backupQuizAnswers = Document(backupDB, "quizAnswers")
-    backupStarboard = Document(backupDB, "starboard")
+        if message.guild:
+            try:
+                guild_config = await self.config.find(guild_id)
+                if message.channel.id in guild_config['ignored_channels']:
+                    return None
+            except (exceptions.IdNotFound, KeyError):
+                pass
+        # Whenever the bot is tagged, respond with its prefix
+        if match := mention.match(message.content):
+            if int(match.group('id')) == (self.user.id):
+                data = await self.config._Document__get_raw(guild_id)
+                if (not data) or ('prefix' not in data):
+                    prefix = self.DEFAULTPREFIX
+                else:
+                    prefix = data['prefix']
+
+                content = 'My prefix here is `%s`' % (prefix)
+                await message.channel.send(content, delete_after=15)
+
+        await self.process_commands(message)
+
+    def mongo(self, collection):
```

Comment:
If you are going to introduce a change in the way database documents are interacted with, please do so throughout the whole bot.

I am also not sure how I feel about this change